docs/arch/adr-010-new-versioning-scheme.md

```diff
@@ -0,0 +1,193 @@
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27"
+```
+_We have decided to transition to the newer **v1beta2** variant. Existing users who prefer to continue working with the outdated format may still do so at their own discretion_
+
+#### Pinning the kURL version
+
+Another feature we aim to offer is the ability for users to fix (pin) their installation to a particular kURL installer version. The organisation of kURL installer releases follows the pattern `YYYY.MM.DD-x`. A complete list of kURL versions can be seen [here](https://github.com/replicatedhq/kURL/releases).
+
+For those seeking to pin their setup to a designated kURL installer version, the `installerVersion` field must be appropriately configured as shown in the example below.
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1beta2"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27"
+  installerVersion: "v2023.04.24-0"
+```
+
+While it's certainly possible to omit the `installerVersion` configuration, doing so means relying upon the current most up-to-dated kURL installer version.
+
+### Add-on versions
+
+In order to maintain greater control over the installation process, we have decided to modify our approach to add-on selection. Going forward, we will no longer allow users to freely select any add-on on any version of their choosing. Instead, we will provide users with specific bundles of add-ons (from here on this ADR will call them _Bundles_), each of which will be available in different versions within a designated Channel.
+
+This is an example of a Bundle on its v1.0.0 version:
+
+| Add-on     | Version              |
+|------------|----------------------|
+| Containerd | 1.6.20               |
+| Flannel    | 0.21.5               |
+| OpenEBS    | 3.5.0.               |
+| Minio      | 2023-03-13T19-46-17Z |
+
+
+To facilitate the selection of a Bundle within a Channel, we will implement a new feature that enables users to pin a specific version of a Bundle. This means that users can select and install the precise version of a Bundle that best aligns with their needs, providing them with reproducibility.
+
+To illustrate the concept of Bundle version within a Channel, we have provided an example of one that could be available within the Channel v1.24. Let's call this _Bundle v1.0.0_:
+
+| Add-on     | Version              |
+|------------|----------------------|
+| Containerd | 1.6.16               |
+| Flannel    | 0.21.4               |
+| OpenEBS    | 3.3.0.               |
+| Minio      | 2023-03-13T19-46-17Z |
+
+For users who wish to pin their Bundle version within a Channel, we will implemented a straightforward solution. By providing the `bundleVersion` parameter as follows, users can easily specify the exact version of the add-on Bundle that they wish to install:
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1beta2"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.24"
+  bundleVersion: "1.0.0"
```

Comment:
Would we have installerVersion and bundleVersion?
I think we need a installerVersion only.
In the future of KURL I imagine the CLI (brainstorming-kurl) applying k0s and then our kURL Operator (brainstorming-ekco). Therefore, the installer release could be tag version of the CLI. Then , we know what ekco as any other dep component which would be installed within.

@emosbaugh @laverya WDYT

Reply:
yes this is what i was thinking

Reply:
It seems like we are converging into removing the "bundle version". I am resolving this conversation as it is duplicate of https://github.com/replicatedhq/kURL/pull/4430#pullrequestreview-1402221415. Please add your input there.